calico_node/startup/startup.go

```diff
@@ -347,19 +370,25 @@ func parseIPEnvironment(envName, envValue string, version int) *net.IPNet {
 	}
 	message("Using IPv%d address from environment: %s=%s", ip.Version(), envName, envValue)
 
-	return ip
+	return ip.String()
 }
 
 // validateIP checks that the IP address is actually on one of the host
 // interfaces and warns if not.
-func validateIP(ipn *net.IPNet) {
+func validateIP(ipn string) {
 	// No validation required if no IP address is specified.
-	if ipn == nil {
+	if ipn == "" {
 		return
 	}
 
+	_, cidr, err := net.ParseCIDR(ipn)
+	if err != nil {
+		fatal("Failed to parse autodetected CIDR '%s'", ipn)
+		terminate()
+	}
+
 	// Pull out the IP as a net.IP (it has useful string and version methods).
-	ip := net.IP{ipn.IP}
+	ip := net.IP{cidr.IP}
```

Comment:
I think you just want to use the IP that is returned as the first parameter of the Parse above.  I think the IP for this parsed CIDR will be masked by the network.